modules/cnf-topology-aware-lifecycle-manager-precache-user-spec-images.adoc

```diff
@@ -0,0 +1,181 @@
+<2> Specifies the minimum required disk space on the cluster in Gigabytes. If unspecified, {cgu-operator} defines a default value for {product-title} images.
+<3> Specifies the list of additional images to pre-cache.
+
+[id="talm-prechache-user-specified-images-preparing-crs_{context}"]
+== Creating the custom resources for pre-caching
+
+You must create the `PreCachingConfig` CR before or concurrently with the `ClusterGroupUpgrade` CR.
+
+. Create the `PreCachingConfig` CR with the list of additional images you want to pre-cache.
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: PreCachingConfig
+metadata:
+  name: exampleconfig
+  namespace: default <1>
+spec:
+  overrides:
+    platformImage: quay.io/openshift-release-dev/ocp-release@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    operatorsIndexes:
+      - registry.example.com:5000/custom-redhat-operators:1.0.0
+    operatorsPackagesAndChannels:
+      - local-storage-operator: stable
+      - ptp-operator: stable
+      - sriov-network-operator: stable
+  spaceRequired: 30Gi
+  excludePrecachePatterns:
+    - aws
+    - vsphere
+  additionalImages:
+    - quay.io/exampleconfig/application1@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    - quay.io/exampleconfig/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+    - quay.io/exampleconfig/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+----
+<1> The `namespace` must be accessible to the hub cluster.
+
+. Create a `ClusterGroupUpgrade` CR with the `preCaching` field set to `true` and specify the `PreCachingConfig` CR created in the previous step:
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu
+  namespace: default
+spec:
+  clusters:
+  - sno1
+  - sno2
+  preCaching: true
+  preCachingConfigRef:
+  - name: exampleconfig
+    namespace: default
+  remediationStrategy:
+    timeout: 240
+----
+
+[WARNING]
+====
+Once you install the images on the cluster, you cannot change or delete them.
+====
+
+. When you want to start pre-caching the images, apply the `ClusterGroupUpgrade` CR by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f cgu.yaml
+----
+
+. {cgu-operator} verifies the `ClusterGroupUpgrade` CR.
+
++
+[NOTE]
+====
+All sites are pre-cached concurrently.
+====
+
+.Verification
+
+You can check the pre-caching status by running the following command:
++
+[source,terminal]
+----
+$ oc get jobs,pods -n openshift-talo-pre-cache
+----
+
+. The pre-caching configurations are validated by checking if the managed policies exist.
+A valid configuration results in the following status:
++
+.Example of validated PreCachingConfig CR
+[source,terminal]
+----
+Type:    "PrecacheSpecValid",
+Status:  True,
+Reason:  "PrecacheSpecIsWellFormed",
+Message: "Precaching spec is valid and consistent"
+----
+
```

Comment:
It would be great if you can also include and invalid config example. Refer to the following PR:  https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/603/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R48 and https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/603/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R48.